lib/elixir/lib/enum.ex

```diff
@@ -294,9 +294,14 @@ defmodule Enum do
       iex> Enum.all?([2, 3, 4], fn x -> rem(x, 2) == 0 end)
       false
 
-      iex> Enum.all?([], fn x -> x > 0 end)
+      iex> Enum.all?([], fn x -> nil end)
       true
 
+  As the last example shows, `Enum.all?/2` returns `true` if `enumerable` is
+  empty, regardless of `fun`. In an empty enumerable there is no element for
+  which `fun` returns a falsy value, so the result must be `true`. This is a
+  standard mathematical argument for empty collections.
```

Comment:
What do you think about?

```suggestion
  which `fun` returns a falsy value, so the result must be `true`. This mimics
  the mathematical definition of similar quantifiers for empty collections.
```

Or at least, we can replace "standard" by "classical" or "well-defined". The issue is that "standard" has multiple meanings and it may not be clear which one we had in mind. In particular, I want to avoid the interpretation of "very familiar" or "well-established", which may lead a reader to think they should be aware with the mathematical interpretation, which is not strictly a requirement.

Another option is to replace mathematical by logical, if you think it is suitable. :)

Reply:
Good point. Let me think.

Reply:
What about 

> This is a standard logical argument for empty collections.

I am looking for something informal and lightweight that resonates to anyone without sounding "academic", to say it somehow.

Reply:
My biggest issue is with the word standard. Can we use one of "well-defined", "accepted", "classical", "common"? My preference is "well-defined" so we keep authoritative but without being too academic, as you said.

In other words:

```suggestion
  which `fun` returns a falsy value, so the result must be `true`. This is a
  well-defined logical argument for empty collections.
```

Reply:
Perfect :).